src/FileUtility.ts

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+import * as fs from 'fs-plus';
+import * as path from 'path';
+import * as sdk from 'vscode-iot-device-cube-sdk';
+import {ScaffoldType} from './constants';
+
+export class FileUtility {
+
+  static async exists(type: ScaffoldType, path: string): Promise<boolean> {
+    if (type === ScaffoldType.local) {
+      return await sdk.FileSystem.exists(path);
+    } else {
+      return new Promise((resolve: (exist: boolean) => void) => {
```

Comment:
I think you could still use fs.statsync in this case to avoid the callback.
You also need to check the return value to detect whether this is a file or a dir.

Reply:
I found there is a directoryExists api below, what does this API plan to detect?

Reply:
The `exists` function can be removed. Fixed.